src/lib/InstantSearch.js

```diff
@@ -395,6 +413,21 @@ See: https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-o
     return this._createURL(this.helper.state.setQueryParameters(params));
   }
 
+  _init(helper, state) {
+    this.widgets.forEach(widget => {
+      if (widget.init) {
+        widget.init({
+          state,
+          helper,
+          templatesConfig: this.templatesConfig,
+          createURL: this._createAbsoluteURL,
+          onHistoryChange: this._onHistoryChange,
+          instantSearchInstance: this,
+        });
+      }
+    });
+  }
+
   _render(helper, results, state) {
```

Comment:
unrelated to this pr, but why do we even bother splitting up the arguments here, isn't it always helper.results & helper.state?

Reply:
It won't be there in the next PRs anyway.